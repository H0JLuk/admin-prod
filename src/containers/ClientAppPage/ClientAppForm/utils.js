import { notification } from 'antd';
import { addSettings, updateSettingsList } from '../../../api/services/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';

export const showNotify = (message, isError) => {
    const config = {
        duration: 5,
        placement: 'bottomRight',
    };

    if (isError) {
        notification.error({
            ...config,
            message,
        });
    } else {
        notification.success({
            ...config,
            message,
        });
    }

};

export async function createOrUpdateKey (changedParams) {
    const updateSettings = [];
    const addSettingsArr = [];

    changedParams.forEach(({ type, ...params }) => {
        type === SETTINGS_TYPES.EDIT
            ? updateSettings.push(params)
            : addSettingsArr.push(params);
    });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export const checkExistDesignSettings = (settings) => {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
};

export const getImage = (imageName) => {
    return `${process.env.PUBLIC_URL}/images/clientAppDesign/${imageName}.png`;
};


