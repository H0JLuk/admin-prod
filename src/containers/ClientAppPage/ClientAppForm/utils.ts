import { customNotifications } from '@utils/notifications';
import { ArgsProps } from 'antd/lib/notification';
import { addSettings, updateSettingsList } from '@apiServices/settingsService';
import { designKeysForCheck, SETTINGS_TYPES } from './ClientAppFormConstants';
import { ISettingObject, SettingDto } from '@types';
import ROLES from '@constants/roles';

export interface IChangedParam extends Pick<SettingDto, 'clientAppCode' | 'key' | 'value' | 'userRole'> {
    type: SETTINGS_TYPES;
}

export const showNotify = (message: React.ReactNode, isError?: boolean) => {
    const config: ArgsProps = {
        duration: 5,
        message,
    };

    if (isError) {
        customNotifications.error(config);
    } else {
        customNotifications.success(config);
    }

};

type UpdateAndAddSettings = {
    updateSettings: SettingDto[];
    addSettingsArr: SettingDto[];
};

export async function createOrUpdateKey(changedParams: IChangedParam[]) {
    const { updateSettings, addSettingsArr } = changedParams.reduce<UpdateAndAddSettings>((result, { type, ...params }) => {
        const key = type === SETTINGS_TYPES.EDIT ? 'updateSettings' : 'addSettingsArr';

        if (params.key === 'referralTokenLifetime') {
            delete params.clientAppCode;
            params.key = 'token_lifetime';
            params.userRole = ROLES.REFERAL_LINK;
        }

        return {
            ...result,
            [key]: [...result[key], params],
        };
    }, { updateSettings: [], addSettingsArr: [] });

    updateSettings.length && (await updateSettingsList(updateSettings));
    addSettingsArr.length && (await addSettings(addSettingsArr));
}

export function checkExistDesignSettings(settings: ISettingObject) {
    return designKeysForCheck.some(key => !Object.prototype.hasOwnProperty.call(settings, key));
}

export function getImage(imageName: string) {
    return `${process.env.PUBLIC_URL}/themes/${imageName}/main-page-image.svg`;
}


