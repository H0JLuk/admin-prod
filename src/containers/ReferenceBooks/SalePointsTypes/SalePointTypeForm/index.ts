import SalePointTypeForm from './SalePointTypeForm';

export default SalePointTypeForm;
