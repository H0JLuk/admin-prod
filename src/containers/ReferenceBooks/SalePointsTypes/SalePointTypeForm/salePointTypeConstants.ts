export const NEW_SALE_POINT_TYPE_TITLE = 'Новый тип точки продажи';
export const EDIT_MODE = 'edit';
export const ON_DELETE_MESSAGE = 'Вы уверены, что хотите удалить тип точки продажи';
export const SALE_POINT_TYPE_FORM_ID = 'sale-point-type';
export const SALE_POINT_TYPE_PRIORITY_ERROR = 'Приоритет типа точки продажи должен быть числом от 1 до 999';

export const SALE_POINT_TYPE_FIELD = {
    NAME: {
        name: 'name',
        label: 'Название типа точки продаж',
        placeholder: 'Название',
    },
    DESCRIPTION: {
        name: 'description',
        label: 'Описание типа точки продаж',
        placeholder: 'Описание',
    },
    PRIORITY: {
        name: 'priority',
        label: 'Приоритет типа точки продаж',
        placeholder: 'Приоритет',
    },
    KIND: {
        name: 'kind',
        label: 'Вид точки продаж',
    },
};
