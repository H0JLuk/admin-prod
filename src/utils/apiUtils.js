const cache = {};

export const getRequestCacheDecorator = (Api, url, params, responseType) => {
    if (cache[url] !== undefined) {
        return Promise.resolve(cache[url]);
    }
    return Api.get(url, params, responseType).then(response => {
        return cache[url] = response;
    });
};
