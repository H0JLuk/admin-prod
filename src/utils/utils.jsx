
import { ExclamationCircleOutlined } from '@ant-design/icons';
import { Modal } from 'antd';
import React from 'react';

export function highlightSearchingText(value = '', searchValue = '', highlightClassName) {
    if (!searchValue || !value) {
        return <span>{ value }</span>;
    }

    const newValue = [];
    const index = value
        .trim()
        .toLocaleUpperCase()
        .indexOf(
            searchValue
                .trim()
                .toLocaleUpperCase()
            );

    if (index + 1) {
        newValue.push(value.substring(0, index));
        newValue.push(
            <span key={ index } className={ highlightClassName }>
                { value.substring(index, index + searchValue.length) }
            </span>
        );
        newValue.push(value.substring(index + searchValue.length));
    } else {
        newValue.push(value);
    }

    return <span>{ newValue }</span>;
}

export function getStringOptionValue({ parentName = '', name = '' } = {}) {
    if (!parentName) {
        return name ?? '';
    }

    return [parentName, name].join(', ');
}

export function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function confirmModal({
    onOk,
    title = 'Вы уверены?',
    icon: customIcon,
    content = '',
    okText = 'Да',
    cancelText = 'Отмена',
    centered = true,
    ...restOptions
}) {
    const icon = customIcon ? customIcon : <ExclamationCircleOutlined />;
    Modal.confirm({ onOk, title, icon, content, okText, cancelText, centered, ...restOptions });
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function successModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}) {
    Modal.success({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}

/**
 * @param {import('antd/lib/modal').ModalFuncProps} data
 */
export function errorModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}) {
    Modal.error({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}