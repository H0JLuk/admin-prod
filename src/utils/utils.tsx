import React from 'react';
import { DEFAULT_SLEEP_TIME } from '@constants/common';
import { Modal, ModalFuncProps } from 'antd';
import { ExclamationCircleOutlined } from '@ant-design/icons';

export function highlightSearchingText(value = '', searchValue = '', highlightClassName = '') {
    if (!searchValue || !value) {
        return <span>{value}</span>;
    }

    const newValue = [];
    const index = value
        .trim()
        .toLocaleUpperCase()
        .indexOf(
            searchValue
                .trim()
                .toLocaleUpperCase(),
        );

    if (index + 1) {
        newValue.push(value.substring(0, index));
        newValue.push(
            <span key={index} className={highlightClassName}>
                {value.substring(index, index + searchValue.length)}
            </span>,
        );
        newValue.push(value.substring(index + searchValue.length));
    } else {
        newValue.push(value);
    }

    return <span>{newValue}</span>;
}

export async function requestsWithMinWait<T>(promise: Promise<T>, minWait = DEFAULT_SLEEP_TIME): Promise<T> {
    const [result] = await Promise.all([promise, sleep(minWait)]);
    return result;
}

export function getStringOptionValue({ parentName = '', name = '' } = {}) {
    if (!parentName) {
        return name ?? '';
    }

    return [parentName, name].join(', ');
}

export function sleep(ms: number): Promise<void> {
    return new Promise(resolve => setTimeout(resolve, ms));
}

export function confirmModal({
    onOk,
    title = 'Вы уверены?',
    icon: customIcon,
    content = '',
    okText = 'Да',
    cancelText = 'Отмена',
    centered = true,
    ...restOptions
}: ModalFuncProps) {
    const icon = customIcon ? customIcon : <ExclamationCircleOutlined />;
    Modal.confirm({ onOk, title, icon, content, okText, cancelText, centered, ...restOptions });
}

export function successModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}: ModalFuncProps) {
    Modal.success({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}

export function errorModal({
    onOk,
    title = 'Готово',
    okText = 'ОК',
    cancelText = 'Отмена',
    centered = true,
    cancelButtonProps = { hidden: true },
    maskClosable = false,
    ...restOptions
}: ModalFuncProps) {
    Modal.error({ onOk, title, okText, cancelText, centered, cancelButtonProps, maskClosable, ...restOptions });
}
