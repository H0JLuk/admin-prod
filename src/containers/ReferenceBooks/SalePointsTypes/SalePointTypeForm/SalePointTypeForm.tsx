import React, { useState, useEffect, useRef } from 'react';
import { RouteComponentProps, useLocation, useParams } from 'react-router-dom';
import { Col, Form, Input, message, Row, Radio } from 'antd';

import Loading from '@components/Loading';
import { confirmModal } from '@utils/utils';
import { FORM_RULES, getPatternAndMessage } from '@utils/validators';
import { SalePointType, SaveSalePointTypeRequest } from '@types';
import ReferenceButtons from '@containers/ReferenceBooks/ReferenceButtons';
import { showNotify } from '@containers/ClientAppPage/ClientAppForm/utils';
import { BUTTON_TEXT, SALE_POINT_TYPE, SALE_POINT_TYPE_RU } from '@constants/common';
import {
    addSalePointType,
    deleteSalePointType,
    editSalePointType,
    getSalePointTypeById,
} from '@apiServices/salePointService';
import {
    EDIT_MODE,
    NEW_SALE_POINT_TYPE_TITLE,
    ON_DELETE_MESSAGE,
    SALE_POINT_TYPE_FIELD,
    SALE_POINT_TYPE_FORM_ID,
    SALE_POINT_TYPE_PRIORITY_ERROR,
} from './salePointTypeConstants';

import styles from './SalePointTypeForm.module.css';

type SalePointTypeFormProps = RouteComponentProps & {
    matchPath: string;
    mode: string;
};

type SalePointTypeState = {
    salePoint?: SalePointType;
} | undefined;

const SALE_POINT_KIND_OPTIONS = [
    { label: SALE_POINT_TYPE_RU.INTERNAL, value: SALE_POINT_TYPE.INTERNAL },
    { label: SALE_POINT_TYPE_RU.EXTERNAL, value: SALE_POINT_TYPE.EXTERNAL },
];

const SalePointTypeForm: React.FC<SalePointTypeFormProps> = ({ matchPath, mode, history }) => {
    const { state, pathname } = useLocation<SalePointTypeState>();
    const params = useParams<{ salePointTypeId: string; }>();
    const salePointType = useRef({} as SalePointType | undefined);
    const [loading, setLoading] = useState(true);
    const [form] = Form.useForm();

    const salePointTypeId = +params.salePointTypeId;
    const isEdit = mode === EDIT_MODE;

    useEffect(() => {
        (async () => {
            if (isEdit) {
                salePointType.current = state?.salePoint || {} as SalePointType;
                if (!state?.salePoint) {
                    salePointType.current = await getSalePointTypeById(salePointTypeId);
                }
                form.setFieldsValue(salePointType.current);
            }
            setLoading(false);
        })();
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    const redirectToList = () => history.push(matchPath);

    if (isEdit && !salePointType.current) {
        message.error(`Тип точки продажи с id ${salePointTypeId} отсутствует`);
        redirectToList();
        return null;
    }

    const onDeleteSalePointType = async () => {
        try {
            await deleteSalePointType(salePointTypeId);
            showNotify(<div>Тип точки продажи <b>{salePointType.current?.name}</b> успешно удален</div>);
            redirectToList();
        } catch (error) {
            if (error.message) {
                message.error(`Ошибка удаления типа точки продажи: ${ error.message }`);
            }
            console.warn(error);
        }
    };

    const handleDelete = () => {
        confirmModal({
            onOk: onDeleteSalePointType,
            title: <b>{ON_DELETE_MESSAGE} <i>{salePointType.current?.name}</i>?</b>,
            okText: BUTTON_TEXT.DELETE,
            cancelText: BUTTON_TEXT.CANCEL,
        });
    };

    const onFinish = async (formData: SaveSalePointTypeRequest) => {
        try {
            setLoading(true);
            if (!isEdit) {
                await addSalePointType(formData);
                showNotify(<div>Тип точки продажи <b>{formData.name}</b> успешно добавлен</div>);
                history.replace(pathname);
            } else {
                await editSalePointType(salePointTypeId, formData);
                showNotify(<div>Тип точки продажи <b>{formData.name}</b> успешно отредактирован</div>);
            }
            redirectToList();
        } catch (error) {
            if (error.message) {
                message.error(`Ошибка ${ isEdit ? 'изменения' : 'создания' } типа точки продажи: ${ error.message }`);
            }
            console.warn(error);
            setLoading(false);
        }
    };

    return (
        <div>
            {loading && <Loading />}
            <div className={styles.wrapper}>
                <div className={styles.pageTitle}>
                    {isEdit ? `Тип точки продажи ${salePointType.current?.name}` : NEW_SALE_POINT_TYPE_TITLE}
                </div>
                <div className={styles.formStyles}>
                    <div className={styles.formWrapper}>
                        <Form
                            form={form}
                            layout="vertical"
                            id={SALE_POINT_TYPE_FORM_ID}
                            onFinish={onFinish}
                            validateTrigger="onSubmit"
                        >
                            <Row gutter={24}>
                                <Col span={12}>
                                    <Form.Item
                                        label={SALE_POINT_TYPE_FIELD.NAME.label}
                                        name={SALE_POINT_TYPE_FIELD.NAME.name}
                                        rules={[
                                            FORM_RULES.REQUIRED,
                                            getPatternAndMessage('salePoint', 'name'),
                                        ]}
                                    >
                                        <Input
                                            placeholder={SALE_POINT_TYPE_FIELD.NAME.placeholder}
                                            allowClear
                                            maxLength={50}
                                        />
                                    </Form.Item>
                                    <Form.Item
                                        label={SALE_POINT_TYPE_FIELD.DESCRIPTION.label}
                                        name={SALE_POINT_TYPE_FIELD.DESCRIPTION.name}
                                        rules={[getPatternAndMessage('salePoint', 'description')]}
                                    >
                                        <Input.TextArea
                                            placeholder={SALE_POINT_TYPE_FIELD.DESCRIPTION.placeholder}
                                            maxLength={300}
                                        />
                                    </Form.Item>
                                </Col>
                                <Col span={12}>
                                    <Form.Item
                                        label={SALE_POINT_TYPE_FIELD.PRIORITY.label}
                                        name={SALE_POINT_TYPE_FIELD.PRIORITY.name}
                                        rules={[
                                            FORM_RULES.REQUIRED,
                                            {
                                                ...FORM_RULES.NUMBER,
                                                min: 1,
                                                max: 999,
                                                message: SALE_POINT_TYPE_PRIORITY_ERROR,
                                            },
                                        ]}
                                    >
                                        <Input
                                            placeholder={SALE_POINT_TYPE_FIELD.PRIORITY.placeholder}
                                            maxLength={3}
                                            allowClear
                                        />
                                    </Form.Item>
                                    <Form.Item
                                        label={SALE_POINT_TYPE_FIELD.KIND.label}
                                        name={SALE_POINT_TYPE_FIELD.KIND.name}
                                        initialValue={salePointType.current?.kind || SALE_POINT_TYPE.INTERNAL}
                                    >
                                        <Radio.Group options={SALE_POINT_KIND_OPTIONS} />
                                    </Form.Item>
                                </Col>
                            </Row>
                        </Form>
                    </div>
                </div>
                <div className={styles.buttonsContainer}>
                    <ReferenceButtons
                        mode={mode}
                        onCancel={redirectToList}
                        onDelete={handleDelete}
                        form={SALE_POINT_TYPE_FORM_ID}
                    />
                </div>
            </div>
        </div>
    );
};

export default SalePointTypeForm;
