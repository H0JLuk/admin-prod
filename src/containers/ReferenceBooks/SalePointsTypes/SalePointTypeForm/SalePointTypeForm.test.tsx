import React from 'react';
import { Location } from 'history';
import { message, Form } from 'antd';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import { match, useLocation } from 'react-router-dom';

import { sleep } from '@setupTests';
import SalePointTypeForm from './SalePointTypeForm';
import {
    getSalePointTypeById,
    addSalePointType,
    editSalePointType,
    deleteSalePointType,
} from '@apiServices/salePointService';

import * as utils from '@utils/utils';

(utils.confirmModal as jest.Mock) = jest.fn();
(message.error as jest.Mock) = jest.fn();

jest.mock('react-router-dom', () => ({
    useLocation: jest.fn(),
    useParams: () => ({ salePointTypeId: '0' }),
}));

jest.mock('@apiServices/salePointService', () => ({
    getSalePointTypeById: jest.fn(),
    addSalePointType: jest.fn(),
    editSalePointType: jest.fn(),
    deleteSalePointType: jest.fn(),
}));

const mockHistoryPush = jest.fn();

const mockTestSalePointType = {
    state: {
        salePoint: {
            deleted: false,
            description: 'Тестовое описание',
            endDate: null,
            id: 0,
            kind: 'INTERNAL',
            name: 'Тест',
            priority: 1,
            startDate: '2020-01-01',
        },
    },
};

const testFormData = {
    description: 'Тест',
    kind: 'EXTERNAL',
    name: 'Тестовое имя',
    priority: 1,
};

const TEST_PROPS = {
    matchPath: 'test',
    mode: 'add',
    history:  {
        push: mockHistoryPush,
        replace: jest.fn(),
    } as any,
    location: {} as Location,
    match: {} as match,
};

describe('<SalePointTypeForm /> test', () => {
    it('should match snapshot', () => {
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const Component = shallow(<SalePointTypeForm {...TEST_PROPS} />);
        expect(Component.html()).toMatchSnapshot();
    });

    it('sale point type should be created', async () => {
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const EditableComponent = mount(<SalePointTypeForm {...TEST_PROPS} />);
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        const formInstance = EditableComponent.find(Form).prop('form')!;
        formInstance.setFieldsValue(testFormData);
        formInstance.submit();
        await sleep();
        expect(addSalePointType).toHaveBeenCalledTimes(1);
        expect(addSalePointType).toHaveBeenCalledWith(testFormData);
    });

    it('should open edit mode', () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        expect(EditableComponent.find('.pageTitle').text()).toBe('Тип точки продажи Тест');
    });

    it('should get sale point from server', async () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => ({}));
        await act(async () => {
            mount(<SalePointTypeForm {...newProps} />);
        });
        expect(getSalePointTypeById).toHaveBeenCalledTimes(1);
        expect(getSalePointTypeById).toHaveBeenCalledWith(0);
    });

    it('sale point type should be edited', async () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        const formData = {
            description: 'Тестовое описание',
            kind: 'EXTERNAL',
            name: 'Тест имени',
            priority: 1,
        };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        const formInstance = EditableComponent.find(Form).prop('form')!;
        formInstance.setFieldsValue(formData);
        formInstance.submit();
        await sleep();
        expect(editSalePointType).toHaveBeenCalled();
    });

    it('sale point type should not be edited', async () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        const formData = {
            description: 'тест некорректного описания&',
            kind: 'EXTERNAL',
            name: 'test',
            priority: 1,
        };
        const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => ({}));
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        (editSalePointType as jest.Mock).mockRejectedValue('Error');
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        const formInstance = EditableComponent.find(Form).prop('form')!;
        formInstance.setFieldsValue(formData);
        formInstance.submit();
        await sleep();
        expect(warnSpy).toHaveBeenCalledTimes(2);
    });

    it('sale point type should not be edited with error message', async () => {
        const formData = {
            description: 'Тестовое описание',
            kind: 'EXTERNAL',
            name: 'Тестовое имя',
            priority: 1,
        };
        console.warn = jest.fn(() => '');
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        (editSalePointType as jest.Mock).mockRejectedValue(new Error('Error'));
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
        const formInstance = EditableComponent.find(Form).prop('form')!;
        formInstance.setFieldsValue(formData);
        formInstance.submit();
        await sleep();
        expect(message.error).toHaveBeenCalledWith('Ошибка изменения типа точки продажи: Error');
    });

    it('should redirect to sale point type list', async () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        EditableComponent.find('button').at(1).simulate('click');
        await sleep();
        expect(mockHistoryPush).toHaveBeenCalled();
    });

    it('sale point type should be delete', async () => {
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        EditableComponent.find('button').at(2).simulate('click');
        await sleep();
        expect(utils.confirmModal).toHaveBeenCalled();
        await (utils.confirmModal as jest.Mock).mock.calls[0][0].onOk();
        expect(deleteSalePointType).toHaveBeenCalled();
    });

    it('sale point type should not be delete', async () => {
        const warnSpy = jest.spyOn(console, 'warn').mockImplementation(() => ({}));
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        (deleteSalePointType as jest.Mock).mockRejectedValue('Error');
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        EditableComponent.find('button').at(2).simulate('click');
        await sleep();
        await (utils.confirmModal as jest.Mock).mock.calls[0][0].onOk();
        expect(warnSpy).toHaveBeenCalledTimes(1);
    });

    it('sale point type should not be delete with error message', async () => {
        console.warn = jest.fn(() => '');
        const newProps = { ...TEST_PROPS, mode: 'edit' };
        (useLocation as jest.Mock).mockImplementation(() => mockTestSalePointType);
        (deleteSalePointType as jest.Mock).mockRejectedValue(new Error('Error'));
        const EditableComponent = mount(<SalePointTypeForm {...newProps} />);
        EditableComponent.find('button').at(2).simulate('click');
        await sleep();
        await (utils.confirmModal as jest.Mock).mock.calls[0][0].onOk();
        expect(message.error).toHaveBeenCalledWith('Ошибка удаления типа точки продажи: Error');
    });
});
