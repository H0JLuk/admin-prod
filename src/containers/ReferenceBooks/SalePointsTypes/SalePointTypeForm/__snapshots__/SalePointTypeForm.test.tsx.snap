// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SalePointTypeForm /> test should match snapshot 1`] = `"<div><div class=\\"loadingContainer\\" data-testid=\\"loader\\"><div class=\\"loading\\"><span role=\\"img\\" aria-label=\\"sync\\" class=\\"anticon anticon-sync anticon-spin\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"sync\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M168 504.2c1-43.7 10-86.1 26.9-126 17.3-41 42.1-77.7 73.7-109.4S337 212.3 378 195c42.4-17.9 87.4-27 133.9-27s91.5 9.1 133.8 27A341.5 341.5 0 01755 268.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.7 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c0-6.7-7.7-10.5-12.9-6.3l-56.4 44.1C765.8 155.1 646.2 92 511.8 92 282.7 92 96.3 275.6 92 503.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8zm756 7.8h-60c-4.4 0-7.9 3.5-8 7.8-1 43.7-10 86.1-26.9 126-17.3 41-42.1 77.8-73.7 109.4A342.45 342.45 0 01512.1 856a342.24 342.24 0 01-243.2-100.8c-9.9-9.9-19.2-20.4-27.8-31.4l60.2-47a8 8 0 00-3-14.1l-175.7-43c-5-1.2-9.9 2.6-9.9 7.7l-.7 181c0 6.7 7.7 10.5 12.9 6.3l56.4-44.1C258.2 868.9 377.8 932 512.2 932c229.2 0 415.5-183.7 419.8-411.8a8 8 0 00-8-8.2z\\"></path></svg></span></div></div><div class=\\"wrapper\\"><div class=\\"pageTitle\\">Новый тип точки продажи</div><div class=\\"formStyles\\"><div class=\\"formWrapper\\"><form id=\\"sale-point-type\\" class=\\"ant-form ant-form-vertical\\"><div class=\\"ant-row\\" style=\\"margin-left:-12px;margin-right:-12px\\"><div style=\\"padding-left:12px;padding-right:12px\\" class=\\"ant-col ant-col-12\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"name\\" class=\\"ant-form-item-required\\" title=\\"Название типа точки продаж\\">Название типа точки продаж</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper\\"><input type=\\"text\\" placeholder=\\"Название\\" maxLength=\\"50\\" value=\\"\\" id=\\"name\\" class=\\"ant-input\\"/><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span></span></span></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"description\\" class=\\"\\" title=\\"Описание типа точки продаж\\">Описание типа точки продаж</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><textarea placeholder=\\"Описание\\" id=\\"description\\" class=\\"ant-input\\"></textarea></div></div></div></div></div><div style=\\"padding-left:12px;padding-right:12px\\" class=\\"ant-col ant-col-12\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"priority\\" class=\\"ant-form-item-required\\" title=\\"Приоритет типа точки продаж\\">Приоритет типа точки продаж</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper\\"><input type=\\"text\\" placeholder=\\"Приоритет\\" maxLength=\\"3\\" value=\\"\\" id=\\"priority\\" class=\\"ant-input\\"/><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span></span></span></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label for=\\"kind\\" class=\\"\\" title=\\"Вид точки продаж\\">Вид точки продаж</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-radio-group ant-radio-group-outline\\" id=\\"kind\\"><label class=\\"ant-radio-wrapper ant-radio-wrapper-checked\\"><span class=\\"ant-radio ant-radio-checked\\"><input type=\\"radio\\" class=\\"ant-radio-input\\" checked=\\"\\" value=\\"INTERNAL\\"/><span class=\\"ant-radio-inner\\"></span></span><span>Внутренняя</span></label><label class=\\"ant-radio-wrapper\\"><span class=\\"ant-radio\\"><input type=\\"radio\\" class=\\"ant-radio-input\\" value=\\"EXTERNAL\\"/><span class=\\"ant-radio-inner\\"></span></span><span>Внешняя</span></label></div></div></div></div></div></div></div></form></div></div><div class=\\"buttonsContainer\\"><button type=\\"button\\" class=\\"ant-btn\\"><span>Отменить</span></button><button form=\\"sale-point-type\\" type=\\"submit\\" class=\\"ant-btn ant-btn-primary\\"><span>Добавить</span></button></div></div></div>"`;
